From 10198da5ec65cb83b0235d14b047054fb25bb48b Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Thu, 23 Apr 2026 00:51:29 +0300
Subject: [PATCH 06/14] Revert "HACK: proton: winedbg: Support dumping crash
 logs to a directory"

This reverts commit aa561aa3b465163a0a632b2fe13cd36297a7fbab.
---
 programs/winedbg/debugger.h   |  1 -
 programs/winedbg/tgt_active.c | 45 -----------------------------------
 programs/winedbg/winedbg.c    |  9 ++++---
 3 files changed, 4 insertions(+), 51 deletions(-)

diff --git a/programs/winedbg/debugger.h b/programs/winedbg/debugger.h
index f3b02e9c5df..ae181a78196 100644
--- a/programs/winedbg/debugger.h
+++ b/programs/winedbg/debugger.h
@@ -311,7 +311,6 @@ extern	DWORD	                dbg_curr_tid;
 extern  dbg_ctx_t               dbg_context;
 extern  BOOL                    dbg_interactiveP;
 extern  HANDLE                  dbg_houtput;
-extern  HANDLE                  dbg_crash_report_file;
 extern  BOOL                    dbg_use_wine_dbg_output;
 
 struct dbg_internal_var
diff --git a/programs/winedbg/tgt_active.c b/programs/winedbg/tgt_active.c
index a8be58d88c3..9f36c907d50 100644
--- a/programs/winedbg/tgt_active.c
+++ b/programs/winedbg/tgt_active.c
@@ -22,8 +22,6 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
-#include <time.h>
-#include <sys/stat.h>
 
 #include "debugger.h"
 #include "psapi.h"
@@ -829,48 +827,6 @@ static HANDLE create_temp_file(void)
                         NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL | FILE_FLAG_DELETE_ON_CLOSE, 0 );
 }
 
-static HANDLE create_crash_report_file(void)
-{
-    const char *dir = getenv("WINE_CRASH_REPORT_DIR");
-    const char *sgi;
-    char timestr[32];
-    char name[MAX_PATH], *c;
-    time_t t;
-    struct tm lt;
-
-    if(!dir || dir[0] == 0)
-        return INVALID_HANDLE_VALUE;
-
-    strcpy(name, dir);
-
-    for(c = name + 1; *c; ++c){
-        if(*c == '/'){
-            *c = 0;
-            CreateDirectoryA(name, NULL);
-            *c = '/';
-        }
-    }
-    CreateDirectoryA(name, NULL);
-
-    sgi = getenv("SteamGameId");
-
-    t = time(NULL);
-    lt = *localtime(&t);
-    strftime(timestr, ARRAY_SIZE(timestr), "%Y-%m-%d_%H:%M:%S", &lt);
-
-    /* /path/to/crash/reports/2021-05-18_13:21:15_appid-976310_crash.log */
-    snprintf(name, ARRAY_SIZE(name),
-            "%s%s/%s_appid-%s_crash.log",
-            dir[0] == '/' ? "Z:/" : "",
-            dir,
-            timestr,
-            sgi ? sgi : "0"
-            );
-
-    return CreateFileA( name, GENERIC_WRITE, FILE_SHARE_READ,
-                        NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, 0 );
-}
-
 /******************************************************************
  *		dbg_active_attach
  *
@@ -963,7 +919,6 @@ enum dbg_start dbg_active_auto(int argc, char* argv[])
         break;
     case TRUE:
         dbg_use_wine_dbg_output = TRUE;
-        dbg_crash_report_file = create_crash_report_file();
         break;
     }
 
diff --git a/programs/winedbg/winedbg.c b/programs/winedbg/winedbg.c
index b7c899d55c4..40ceae48937 100644
--- a/programs/winedbg/winedbg.c
+++ b/programs/winedbg/winedbg.c
@@ -82,7 +82,6 @@ DWORD	                dbg_curr_pid = 0;
 dbg_ctx_t               dbg_context;
 BOOL    	        dbg_interactiveP = FALSE;
 HANDLE                  dbg_houtput = 0;
-HANDLE                  dbg_crash_report_file = INVALID_HANDLE_VALUE;
 BOOL                    dbg_use_wine_dbg_output = FALSE;
 
 static struct list      dbg_process_list = LIST_INIT(dbg_process_list);
@@ -97,7 +96,10 @@ static void dbg_outputA(const char* buffer, int len)
     DWORD w, i;
 
     if (dbg_use_wine_dbg_output)
+    {
         __wine_dbg_output(buffer);
+        return;
+    }
 
     while (len > 0)
     {
@@ -112,10 +114,7 @@ static void dbg_outputA(const char* buffer, int len)
             if (len > 0) i = line_pos;  /* buffer is full, flush anyway */
             else break;
         }
-        if (!dbg_use_wine_dbg_output)
-            WriteFile(dbg_houtput, line_buff, i, &w, NULL);
-        if (dbg_crash_report_file != INVALID_HANDLE_VALUE)
-            WriteFile(dbg_crash_report_file, line_buff, i, &w, NULL);
+        WriteFile(dbg_houtput, line_buff, i, &w, NULL);
         memmove( line_buff, line_buff + i, line_pos - i );
         line_pos -= i;
     }
-- 
2.54.0

